scipy/sparse/compressed.py

```diff
@@ -1102,10 +1102,13 @@ def _divide_sparse(self, other):
 
         if np.issubdtype(r.dtype, np.inexact):
             # Eldiv leaves entries outside the combined sparsity
-            # pattern empty, so they must be filled manually. They are
-            # always nan, so that the matrix is completely full.
+            # pattern empty, so they must be filled manually.
+            # Everything outside of other's sparsity is NaN, and everything
+            # inside it is either zero or defined by eldiv.
             out = np.empty(self.shape, dtype=self.dtype)
             out.fill(np.nan)
+            row, col = other.nonzero()
+            out[row, col] = 0
```

Comment:
Ok, sorry, the code here is OK. I see the bug now --- the cause is that the sparsity pattern from the binop is not actually the union of the sparsity patterns, as the operation discards zeros. It was assumed here that the output pattern is the union.